src/terra-dev-site/guide/DockerDevelopmentEnvironment.i.guide.md

```diff
@@ -0,0 +1,91 @@
+## Docker Development Environment
+
+Terra provides a docker development environment that allows developers access to a common environment to develop Terra components regardless of the OS they are developing on.
+
+In each Terra repository there is a docker-compose.yml file that defines several services, most importantly ```dev``` and ```test```.
+
+To build all the services in the docker-compose file, run:
+```
+docker-compose build
+```
+
+### Dev
+The dev service provides a linux environment with node and npm pre-installed. By default the dev service volume mounts the entire code repo into the docker container to enable hot reloading during development.
+
+To get started developing, run:
+
+```bash
+docker-compose up dev
+```
+This will start the docker container for the repo, run npm install (if node_modules are not detected) and start the dev server. Then you're free to use your favorite editor to manipulate the code in the repo.
```

Comment:
> start the dev server 

since we offer two serve tools, lets specify the `npm run start` script.

Reply:
I'm not sure we need to be that specific. To the consumer either way the server starts.

Reply:
I think be important to specify. a lot of people have been catch-up with confusing between the tt-serve and tt-serve-static and do not realize tests are run using tt-serve-static. People have been tt-server render their site correctly then have tt-serve-static in production serve slightly different which breaks tests with the difference between their dev site and the test site isn't obvious for debugging.